### PR TITLE
Change DRPC logs to user configurable level

### PR DESCRIPTION
**Pull request by ELENAGER**
Update the DRPC logging from level verbose to user configurable log level for better understanding and easy debugging in case of large amount of PVCs. All messages with level V(1) are printed, if we are not changing the debug level flag to INFO. Then these messages will disappear from logs.
In order to change flag, it is necessary to add arg  --zap-log-level=info in ramen deploy.

### CODE REVIEW

File: controllers/drplacementcontrol.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func (d *DRPCInstance) startProcessing() bool {
 }
 
 func (d *DRPCInstance) processPlacement() (bool, error) {
-	d.log.Info("Process DRPC Placement", "DRAction", d.instance.Spec.Action)
+	d.log.V(1).Info("Process DRPC Placement", "DRAction", d.instance.Spec.Action)
```
Comment (nirs) [inline controllers/drplacementcontrol.go:87]: Not clear why do we need `.V(1)` when we already use `.Info()`. Maybe this is a limit of the logger we use, but normally when you use `.Info()` the logger logs a message in INFO level and the log is available if the log level is equal or higher than INFO level.

Maybe this code means that this is a DEBUG level log - in this case we should have something like `.Debug()` call to log the issue.

Comment (nirs) [inline controllers/drplacementcontrol.go:87]: Not clear why do we need `.V(1)` when we already use `.Info()`. Maybe this is a limit of the logger we use, but normally when you use `.Info()` the logger logs a message in INFO level and the log is available if the log level is equal or higher than INFO level.

Maybe this code means that this is a DEBUG level log - in this case we should have something like `.Debug()` call to log the issue.

File: controllers/drplacementcontrol.go [hunk 2]
```diff
@@ -1549,7 +1549,7 @@ func (d *DRPCInstance) ensureClusterDataRestored(homeCluster string) (*rmn.Volum
 }
 
 func (d *DRPCInstance) EnsureCleanup(clusterToSkip string) error {
-	d.log.Info("ensuring cleanup on secondaries")
+	d.log.Info("Ensure cleanup on secondaries")
```
Comment (nirs) [inline controllers/drplacementcontrol.go:1552]: I'm not sure this is better, but it is not part of logging level changes. It will be better to do logging text improvements in a separate commit.

File: controllers/drplacementcontrol.go [hunk 3]
```diff
@@ -2223,6 +2221,9 @@ func (d *DRPCInstance) fixupPlacementForFailover() error {
 		primary, secondaries = d.selectCurrentPrimaryAndSecondaries()
 	}
 
+	d.log.Info(fmt.Sprintf("Fixing PlacementRule for failover. Primary (%s), Secondaries (%v)",
+		primary, secondaries))
```
Comment (nirs) [inline controllers/drplacementcontrol.go:2225]: This is a third type of change that not related to log levels, so it should be a separate commit (show more context in logs) with similar changes.

Comment (ELENAGER) [reply controllers/drplacementcontrol.go]: Removed

File: controllers/drplacementcontrol.go (hunk 2 above)
Comment (ELENAGER) [reply controllers/drplacementcontrol.go]: Restored

**Comment by ShyamsundarR** [comment]
@BenamarMk when we discussed log levels a long time back one of the reasoning that you provided was, turning up the log level when an issue occurs would miss the required logs and hence not a very useful troubleshooting option as such (which I agree with BTW).

Considering the same, the original intent here was to reduce the size of logs, and reduce areas where we print entire status or spec sections for various resources (at least as a start). Should we consider levels here, or should we retain Info/Error logs and look at optimizing what we log?

**Comment by ShyamsundarR** [comment]
> @BenamarMk when we discussed log levels a long time back one of the reasoning that you provided was, turning up the log level when an issue occurs would miss the required logs and hence not a very useful troubleshooting option as such (which I agree with BTW).
> 
> Considering the same, the original intent here was to reduce the size of logs, and reduce areas where we print entire status or spec sections for various resources (at least as a start). Should we consider levels here, or should we retain Info/Error logs and look at optimizing what we log?

@ELENAGER I think we can close this PR and attempt cleaning up big logs where we print entire structures, [example](https://github.com/RamenDR/ramen/blob/015ae963b3cbf0454f8e1278883c75e3bc888381/controllers/drplacementcontrol_controller.go#L1540).

Also, we may actually want to see how to preserve logs when they overflow, which is not done in the code as of now and so we lose logs in a few days or when multiple reconciles are in flight, and prioritize the same (there is no issue for it at present).
